app/services/forest_liana/oidc_client_manager.rb

```diff
@@ -2,33 +2,32 @@
 
 module ForestLiana
   class OidcClientManager
-    def self.get_client_for_callback_url(callback_url)
+    def self.get_client
       begin
         configuration = ForestLiana::OidcConfigurationRetriever.retrieve()
         if ForestLiana.forest_client_id.nil?
-          client_data = Rails.cache.read("#{callback_url}-#{ForestLiana.env_secret}-client-data") || nil
+          client_data = Rails.cache.read("#{ForestLiana.env_secret}-client-data") || nil
           if client_data.nil?
             client_credentials = ForestLiana::OidcDynamicClientRegistrator.register({
               token_endpoint_auth_method: 'none',
-              redirect_uris: [callback_url],
               registration_endpoint: configuration['registration_endpoint']
             })
-            client_data = { :client_id => client_credentials['client_id'], :issuer => configuration['issuer'] }
-            Rails.cache.write("#{callback_url}-#{ForestLiana.env_secret}-client-data", client_data)
+            client_data = { :client_id => client_credentials['client_id'], :issuer => configuration['issuer'], :redirect_uri => client_credentials['redirect_uris'][0] }
+            Rails.cache.write("#{ForestLiana.env_secret}-client-data", client_data)
           end
         else
           client_data = { :client_id => ForestLiana.forest_client_id, :issuer => configuration['issuer'] }
```

Comment:
⚠️ This use case is not supported anymore?!

Reply:
`client_data[:redirect_uri]` will return null if we use user set `ForestLiana.forest_client_id`.

Reply:
Should we still support static clientId?

in case of `ForestLiana.forest_client_id` we should add this to client_data
```
:redirect_uri => File.join(ForestLiana.application_url, "/forest/#{CALLBACK_AUTHENTICATION_ROUTE}").to_s
```

Reply:
👍  well done i didn't notice `forest_client_id`